### PR TITLE
self-protect against broken Mapping implementations

### PR DESCRIPTION
**Pull request by crepererum**

### CODE REVIEW

File: sentry_sdk/utils.py [hunk 1]
```diff
@@ -372,11 +372,16 @@ def object_to_json(obj, remaining_depth=4, memo=None):
                 ]
 
             if isinstance(obj, Mapping):
+                try:
```
Comment (untitaker) [inline sentry_sdk/utils.py]: Instead of this, use `capture_internal_exceptions` around the entire function.

Comment (crepererum) [reply sentry_sdk/utils.py]: It's not really an internal exception. It is broken user code somewhere in the stack the shouldn't lead to the fact that the entire stack-capturing breaks. Also I don't really understand what your suggestion is. I still have to return some value from the function and some catch-all value for all possible problems is likely not too helpful.

Comment (untitaker) [reply sentry_sdk/utils.py]: I am suggesting that you replace this:

```python
def object_to_json(...):
   return <body>
```

with:

```python
def object_to_json(...):
    rv = '<broken repr>'
    with capture_internal_exceptions():
        rv = <body>
    return rv
```

Even if it's not our SDK that is buggy but we still want to allow the user to debug this if they really want the values.

**Comment by crepererum** [comment]
should be fixed now

**Comment by untitaker** [comment]
Thanks!

**Pull request merged**